pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -27,6 +27,21 @@ def test_roundtrip_pickle_with_tz(self):
         unpickled = tm.round_trip_pickle(index)
         tm.assert_index_equal(index, unpickled)
 
+    def test_date_accessor_with_tz(self):
```

Comment:
Instead of 2 separate tests, you can have one test `pytest.mark.parameterize` over the timezone aware and naive data. (e.g. `dtype='datetime64[ns, CET]''` and dtype=`None`)